doc/source/rllib/rllib-offline.rst

```diff
@@ -5,437 +5,2081 @@
+                model_config=DefaultModelConfig(
+                    conv_filters=[[16, 4, 2], [32, 4, 2], [64, 4, 2], [128, 4, 2]],
+                    conv_activation="relu",
+                ),
+                inference_only=False,
+                module_class=BCTorchRLModule,
+                catalog_class=BCCatalog,
+                action_space = gym.spaces.Discrete(2),
+                observation_space=gym.spaces.Box(0, 255, (32, 32, 3), np.float32),
+            ),
+        },
+    )
+
+    # Construct your `OfflineData` class instance.
+    offline_data = ImageOfflineData(config)
+
+    # Check, how the data is transformed.
+    batch = offline_data.data.take_batch(128)
+
+    # Construct your `OfflinePreLearner`.
+    offline_prelearner = ImageOfflinePreLearner(
+        config=config,
+        learner=None,
+        spaces=(
+            config.observation_space,
+            config.action_space,
+        ),
+        module_spec=module_spec,
+    )
+
+    # Transform the raw data to `MultiAgentBatch` data.
+    batch = offline_prelearner(batch)
+
+    # Show the transformed batch.
+    print(f"Batch: {batch}")
+
+In the code example provided, you define a custom :py:class:`~ray.rllib.offline.offline_data.OfflineData` class to handle the reading and preprocessing of image data, converting it from a binary encoding format into `numpy` arrays. Additionally, you implement a custom :py:class:`~ray.rllib.offline.offline_prelearner.OfflinePreLearner` to process this data further, transforming it into a learner-ready :py:class:`~ray.rllib.policy.sample_batch.MultiAgentBatch` format. This demonstrates how the entire Offline Data Pipeline can be customized with your own logic. 
+
+.. tip:: Consider this approach carefully: in many cases, fully transforming your data into a suitable format before engaging RLlib's offline RL API can be more efficient. For instance, in the example above, you could preprocess the entire image dataset into `numpy` arrays beforehand and utilize RLlib's default :py:class:`~ray.rllib.offline.offline_data.OfflineData` class for subsequent steps. 
+
+Monitoring
+----------
+To effectively monitor your offline data pipeline, leverage :ref:`Ray Data's built-in monitoring capacities <monitoring-your-workload>`. Focus on ensuring that all stages of your offline data streaming pipeline are actively processing data. Additionally, keep an eye on the Learner instance, particularly the `learner_update_timer`, which should maintain low values - around `0.02` for small models - to indicate efficient data processing and model updates.
+
+.. note:: RLlib doesn't include :ref:`Ray Data <data>`  metrics in its results or display them in `Tensorboard` through :ref:`Ray Tune <tune-main>`'s :py:class:`~ray.tune.logger.tensorboardx.TBXLoggerCallback`. It's strongly recommended to enable the :ref:`Ray Dashboard <observability-getting-started>`, accessible at `127.0.0.1:8265`, for comprehensive monitoring and insights.
```

Comment:
Interesting. Is there any way to funnel these metrics from data into RLlib's metrics loggers somehow? Not top-prio at this point, but would be super nice to have.

Reply:
I doubt that it will be smooth - instead I guess it will produce much overhead. In addition the metrics are  nested such that they can be optimally shown in the Ray Dashboard. 

I am also unsure at this point if these metrics should be in our TensorBoard as they are more system performance metrics than RL metrics. In larger companies these metrics also get checked by System Administrators/Hardware Engineers while the RL metrics get checked by RL engineers. But I am open for discussion.